core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -60,8 +60,10 @@
     public static final ParseField TYPE_FIELD = new ParseField("type");
     public static final ParseField QUERY_FIELD = new ParseField("query");
 
-    /** The default name for the match query */
-    public static final String NAME = "match";
+    /** The names for the match query. */
+    // TODO replace this with a Types like object
+    public static final Names NAMES = new Names("match", "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix",
```

Comment:
I agree with this however, once 1) is fixed we shouldn't need a Names class, but instead should just be able to use a single String constant for the name to be serialised since we don't want different variants being used on the stream anyway. In all cases I would expect `ParseField#getPreferredName()` to return the same name as the string constant for the stream but I'm happy if they are actually separate constants.